test/e2e/specs/demo.test.js

```diff
@@ -46,13 +43,15 @@ describe( 'new editor state', () => {
 		await visitAdmin( 'post-new.php', 'gutenberg-demo' );
 	} );
 
-	it( 'should not error', () => {
-		// This test case is intentionally empty. The `beforeAll` lifecycle of
-		// navigating to the Demo page is sufficient assertion in itself, as it
-		// will trigger the global console error capturing if an error occurs
-		// during this process.
-		//
-		// If any other test cases are added which verify expected behavior of
-		// the demo post, this empty test case can be removed.
+	it( 'content should load without making the post dirty', async () => {
+		await page.waitForFunction( () => {
+			const editor = wp.data.select( 'core/editor' );
+			if ( editor.isEditedPostDirty() ) {
+				// Will cause the test to fail if the post is dirty.
+				// eslint-disable-next-line no-console
+				console.error( 'Demo page was marked dirty' );
```

Comment:
I think we may be able to make this logic more explicit using the evaluate function.
E.g:
```
const isDirty = await page.evaluate( () => {
		const { select } = window.wp.data;
		return select( 'core/editor' ). isEditedPostDirty();
} );
expect( isDirty ).toBeFalsy();
```